src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -97,5 +98,12 @@ export const decodeQrCode = ({
 		return { status: 'token_incompatible' };
 	}
 
+	const amount =
+		functionName === 'transfer'
+			? uint256
+			: nonNullish(value)
+				? +formatEther(value.toString())
```

Comment:
- we do not use `formatEther` anywhere else in the code base, could we potentially use `formatToken` here?

- it's always 18 decimals here even for ERC20 such as USDC?

Reply:
- changing it;
- well, when it's a token that is a ERC20, the amount is equal to `uint256`, but, using `formatToken`, we will provide the `token.decimals` input, to avoid any doubts